indexer/cell_coverer.hpp

```diff
@@ -169,8 +169,8 @@ void CoverSpiral(m2::RectD rect, int maxDepth, vector<CellId> & result)
   };
 
   auto const coordsAreValid = [](pair<int32_t, int32_t> const & xy) {
-    return xy.first >= 0 && xy.second >= 0 && xy.first <= CellId::MAX_COORD &&
-           xy.second <= CellId::MAX_COORD;
+    return xy.first >= 0 && xy.second >= 0 && xy.first <= static_cast<int32_t>(CellId::MAX_COORD) &&
```

Comment:
I'd rather cast xy.first to uint32_t here.

Reply:
Reasoning was that there is already an assert just below that ensures that this cast is safe.
So this seemed the least risky.
I don't really mind though.

Reply:
My reasoning was:
1. This is what happens anyway, an explicit cast to the unsigned type.
2. Imagine that we change MAX_COORD to uint64_t.

As for the assert below, I didn't notice it and I don't even think it is the right way to check what its author wanted to check:
https://pastebin.com/K9ts145N
https://ideone.com/wNg1pa